container/container.go

```diff
@@ -126,14 +141,47 @@ func (c *InfoClient) getContainer(containerId string, useCache bool) (*Container
 	} else {
 		contData.TsuruApp = true
 	}
+
+	if tags, ok := contData.GetLabelAny(logTagLabels...); ok {
+		contData.Tags = strings.Split(tags, ",")
+	}
+	contData.RawExtra = c.genRawExtra(contData.Tags)
+
 	contData.ShortHostname = contData.Config.Hostname
 	if hexRegex.MatchString(contData.Config.Hostname) && len(contData.Config.Hostname) > containerIDTrimSize {
 		contData.ShortHostname = contData.Config.Hostname[:containerIDTrimSize]
 	}
+
 	c.containerCache.Add(containerId, &contData)
 	return &contData, nil
 }
 
+func (c *InfoClient) genRawExtra(tags []string) *json.RawMessage {
+	extra := &c.extra
+	if _tags, ok := c.decodedExtra["_tags"]; ok && len(tags) > 0 {
+		newExtra, err := json.Marshal(map[string]interface{}{"_tags": _tags.(string) + "," + strings.Join(tags, ",")})
+		if err != nil {
+			bslog.Errorf("Unable to join tags: %v", err)
+		} else {
+			raw := json.RawMessage(newExtra)
+			extra = &raw
+		}
+	}
+	return extra
+}
+
+func (c *InfoClient) configGelfExtraTags() {
+	extra := config.StringEnvOrDefault("", "LOG_GELF_EXTRA_TAGS")
+	if extra != "" {
+		c.decodedExtra = map[string]interface{}{}
```

Comment:
Could this be a `map[string]string`? This would avoid the type coercion and possible panic on the `_tags.(string)` above if LOG_GELF_EXTRA_TAGS is invalid.